### ISSUE TITLE
@Retain and @Arg in Fragment inheritance

### ISSUE THREAD

**Issue by ykostova** [tom91136/Akatsuki#18]
If a fragment A has @Retained fields and it is extended by a fragment B with both @Retained and @Arg fields, the latter's (B's) builder cannot be compiled because a Builder cannot be found for the parent (A). 

This situation can be easily recreated by letting ArgRetainedFragment extend BaseRetainedFragment from the provided sample. 

The problem can be resolved by introducing false @Arg fields in the parent (A) fragment but it is my understand that this is not part of the intended behaviour. 

PS: Amazing library ! Thanks for all your hard work.

**Comment by tom91136** [comment]
Confirmed.
I'm currently doing a full rewrite of the library since it's getting very messy, see #17 for background. I'll make sure to add specific test cases for this in the rewrite. Please leave this issue open so I can close it via commit once the rewrite is done.

PS:Sorry if you need this fixed urgently, but the rewrite will take a while as I'm having exams in Uni.

**Comment by ykostova** [comment]
Thanks very much for that. I am excited to see the new version. And again thanks for your effort :)
